server/controllers/request.js

```diff
@@ -0,0 +1,77 @@
+const Request = require("../Models/Request");
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+
+// @route GET /request
+// @desc List of requests for logged in user
+// @access Private
+exports.getRequests = asyncHandler(async (req, res, next) => {
+  const userId = req.user.id;
+
+  const requests = await Request.find({
+    $or: [{ userId: userId }, { sitterId: userId }],
+  });
+
+  res.send({ requests });
+});
+
+// @route POST /request
+// @desc Create a new request
+// @access Private
+exports.createRequest = asyncHandler(async (req, res, next) => {
+  const userId = req.user.id;
+
+  const { sitterId, startDate, endDate } = req.body;
+
+  if (!startDate || !endDate) {
```

Comment:
Optional: Alternatively, you can create middleware functions to check the validity of the `req.body` :)